job-distributor/v1/job/job.proto

```diff
@@ -148,6 +151,22 @@ message ProposeJobResponse {
   Proposal proposal = 1; // Details of the newly created proposal.
 }
 
+// BatchProposeJobRequest contains the job spec and list of nodes to propose it to.
+message BatchProposeJobRequest {
+  repeated string node_ids = 1; // List of node IDs to which the job will be proposed.
+  string spec = 2; // Shared job spec across all nodes.
+  repeated api.label.Label labels = 3; // Shared labels associated with the job.
+}
+
+// BatchProposeJobResponse provides the success and failure statuses for each node in the batch.
+message BatchProposeJobResponse {
+  // Contains the successful job proposals, mapped by node ID.
+  map<string, ProposeJobResponse> success_responses = 1;
+
+  // Contains error details for each failed job proposal, mapped by node ID.
+  map<string, string> failed_responses = 2;
```

Comment:
I wonder instead of `string`, would it be better if it is an object (it can contain a string for now) but this allows us to extend in future with more information instead of a breaking change.

Reply:
Are you referring to the keys, the error strings or both? For errors, using an object would be useful in case we want to extend the context with additional information like the kind of error, not sure about key objects though let me verify what can be do

Reply:
nah not the keys, string is good, but mainly the value for extensibility